OpenRA.Mods.Cnc/Activities/Leap.cs

```diff
@@ -43,7 +45,9 @@ public Leap(Actor self, Actor target, Armament a, WDist speed, WAngle angle, Bit
 			this.angle = angle;
 			this.damageTypes = damageTypes;
 			mobile = self.Trait<Mobile>();
+			notifyMoving = self.TraitsImplementing<INotifyMoving>().ToArray();
 			mobile.SetLocation(mobile.FromCell, mobile.FromSubCell, targetMobile.FromCell, targetMobile.FromSubCell);
+			wasMoving = mobile.IsMoving;
```

Comment:
Changing the actor state from inside the ctor is really bad practice (consider queuing).  Can we move these lines into `OnFirstRun` instead?

Reply:
Done. RA dogs still work, too.